sdk/go/common/resource/asset_test.go

```diff
@@ -539,6 +539,19 @@ func TestFileExtentionSniffing(t *testing.T) {
 	assert.Equal(t, ArchiveFormat(NotArchive), detectArchiveFormat("./some/path/who.even.knows"))
 }
 
+func TestEmptyArchiveRoundTrip(t *testing.T) {
+	t.Parallel()
+
+	emptyArchive, err := NewAssetArchive(nil)
+	assert.True(t, emptyArchive.IsAssets(), "even empty archives should be have empty assets")
+	assert.NoError(t, err, "Creating an empty archive should work")
```

Comment:
A) Do the err check first
B) Use require.NoError, generally if we have an error the other asserts will just end up panic'ing (due to like nil access exceptions and the like)

Reply:
✅